Changed test data generation for SNP data.

gijskant opened the pull request:
Vary the value of minorAllele in SnpDataByProbeCoreDb. This
would detect a previously undetected bug in SnpLzRow
(fixed in d178a3e6d0056492cf94cb1e9ed66dcf472a8656).

gijskant committed:
Changed test data generation for SNP data.

Vary the value of minorAllele in SnpDataByProbeCoreDb. This
would detect a previously undetected bug in SnpLzRow
(fixed in d178a3e6d0056492cf94cb1e9ed66dcf472a8656).

cataphract commented on transmart-core-db-tests/src/groovy/org/transmartproject/db/dataquery/highdim/snp_lz/SnpLzTestData.groovy:
```diff
@@ -208,6 +208,16 @@ class SnpLzTestData {
+            long a2Count = 0
+            gtss.each {
+                a1Count += it.findAll( a1 ).size()
+                a2Count += it.findAll( a2 ).size()
```
I'm not sure what's happening here. Up we see that gtss has the format `['T T', 'T A', ...]`. Then `it` inside the closure must be something like `'T T'`. OTOH, `a1` is a String. So you get something like `'T T'.findAll { 'A' }.size()`, which always returns the number of characters in the first string, that is 3.

This function is also inconsistent: you left `def a1 = annotation.alt` at the top, but then `a1` may not be the alternative. It seems to me that the choice about which allele will be minor one should be made before `a1` and `a2` are assigned.

gijskant replied:
Okay, I'll have another look.
But as far as I understand, minor alleles are not always alternate alleles.

cataphract replied:
I never said that was the case; on the contrary, the comments on lines 201 and 202 concede as much.

gijskant committed:
Change type of chromosome position to Long.

Datatype for chromosome is Number of size 22 in the database,
but was Integer of BigDecimal in the code. Now changed to be
Long.

rnugraha committed:
To fetch data in batch (5000)

cataphract committed:
Merge pull request #4 from thehyve/fix-highdim-service

To fetch data in batch (5000)

cataphract commented:
Incorporated into #6 . Closing
